src/client/datascience/kernel-launcher/kernelProcess.ts

```diff
@@ -437,7 +438,11 @@ export class KernelProcess implements IKernelProcess {
 
             // If we don't have a KernelDaemon here & we're not running a Python module either.
             // The kernelspec argv could be something like [python, main.py, --something, --something-else, -f,{connection_file}]
-            exeObs = executionService.execObservable(this.launchKernelSpec.argv.slice(1), {
+            const args = this.launchKernelSpec.argv.slice(1);
+            if (this.jupyterSettings.enablePythonKernelLogging) {
+                args.push('--debug');
```

Comment:
Need to see what's going on inside the kernel

Reply:
LOL. I have this as the default on my own machine.